packages/forms/src/elements/FauxInput.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { composeEventHandlers } from '@zendeskgarden/container-utilities';
+import { StyledTextFauxInput } from '../styled';
+import VALIDATION from '../utils/validation';
+
+/**
+ * Provides styling without native input backing; accepts all `<div>`
+ * attributes and events.
+ */
+const FauxInput = React.forwardRef(({ onFocus, onBlur, disabled, ...props }, ref) => {
+  const [isFocused, setIsFocused] = useState(false);
+
+  const onFocusHandler = composeEventHandlers(onFocus, () => {
+    setIsFocused(true);
+  });
+
+  const onBlurHandler = composeEventHandlers(onBlur, () => {
+    setIsFocused(false);
+  });
+
+  return (
+    <StyledTextFauxInput
+      onFocus={onFocusHandler}
+      onBlur={onBlurHandler}
+      isFocused={isFocused}
+      isDisabled={disabled}
+      tabIndex={disabled ? null : 0}
+      ref={ref}
+      {...props}
+    />
+  );
+});
+
+FauxInput.propTypes = {
+  /** Apply compact styling */
+  isCompact: PropTypes.bool,
+  /** Remove borders and padding */
+  isBare: PropTypes.bool,
+  /** Apply inset `box-shadow` styling on focus */
+  focusInset: PropTypes.bool,
```

Comment:
This prop doesn't appear to be a native HTML attribute. Is there a particular reason why this prop is not using the `isBoolean` naming format (e.g. `isFocusInset`)?

Reply:
This prop is used throughout: `forms`, `dropdowns`, `buttons`. I'm fine with an improved name (`hasFocusInset`?), but we'll need to agree that this is the right thing to do. Let's discuss. /cc @austin94

Reply:
`hasInsetFocus`?